WooCommerce/Classes/ViewRelated/Products/Variations/Edit Attributes/AttributePickerViewController.swift

```diff
@@ -133,14 +136,39 @@ private extension AttributePickerViewController {
 
 // MARK: - Navigation actions handling
 //
-extension AttributePickerViewController {
+private extension AttributePickerViewController {
+
+    @objc func doneButtonPressed() {
+        onCompletion(viewModel.resultAttributes)
+    }
 
-    @objc private func doneButtonPressed() {
-        onCompletion(variationModel.productVariation.attributes)
+    func presentAttributeOptions(for existingAttribute: ProductAttribute?) {
+        guard let existingAttribute = existingAttribute else {
+            return
+        }
+
+        let oldAttribute = viewModel.selectedOption(for: existingAttribute)
+
+        let command = AttributeOptionListSelectorCommand(attribute: existingAttribute, selectedOption: oldAttribute)
+        let attributeOptionListSelector = ListSelectorViewController(command: command) { [weak self] selected in
+            self?.onAttributeOptionListSelectorCompletion(parentAttribute: existingAttribute, oldAttribute: oldAttribute, selectedRow: selected)
+        }
+        show(attributeOptionListSelector, sender: self)
     }
 
-    private func presentAttributeOptions(for existingAttribute: ProductAttribute) {
-        // TODO-3515: Show options for attribute
+    func onAttributeOptionListSelectorCompletion(parentAttribute: ProductAttribute,
+                                                 oldAttribute: ProductVariationAttribute?,
+                                                 selectedRow: AttributeOptionListSelectorCommand.Row?) {
+        let newAttribute: ProductVariationAttribute?
+        if case .option(let selectedOption) = selectedRow {
+            newAttribute = ProductVariationAttribute(id: parentAttribute.attributeID, name: parentAttribute.name, option: selectedOption)
+        } else {
+            newAttribute = nil
+        }
```

Comment:
I like to use inline closures for adding initialization code like this.
```swift
let newAttribute: ProductVariationAttribute? = {
    switch selectedRow {
        case let .option(selectedOption):
            return ProductVariationAttribute(id: parentAttribute.attributeID, name: parentAttribute.name, option: selectedOption)
        default:
            return nil
    }
}()
```

That way we indicate to the reader that the code inside de closure is only needed for creating `newAttribute` and does not mess with any other part of the method

Reply:
That's perfect for this case, thanks for sharing! 💯